sdk/tests/wallet/output_preparation.rs

```diff
@@ -557,3 +559,192 @@ async fn prepare_nft_output_features_update() -> Result<()> {
+            OutputParams {
+                recipient_address: *address,
+                amount: 100, // leave more than min. deposit
+                assets: None,
+                features: None,
+                unlocks: None,
+                storage_deposit: Some(StorageDeposit {
+                    return_strategy: Some(ReturnStrategy::Return),
+                    use_excess_if_low: Some(true),
+                }),
+            },
+            None,
+        )
+        .await?;
+
+    // Check if the output has enough amount to cover the storage deposit
+    output.verify_storage_deposit(rent_structure, token_supply)?;
+    // We use excess if leftover is too small, so amount == all available balance
+    assert_eq!(output.amount(), 63900);
+    // storage deposit gifted, only address unlock condition
+    assert_eq!(output.unlock_conditions().unwrap().len(), 2);
+    // We have ReturnStrategy::Return, so leftover amount gets returned
+    let sdr = output.unlock_conditions().unwrap().storage_deposit_return().unwrap();
+    assert_eq!(sdr.amount(), 63900 - 100);
+
+    tear_down(storage_path)
+}
+
+#[ignore]
+#[tokio::test]
+async fn prepare_output_only_single_nft() -> Result<()> {
+    let storage_path = "test-storage/prepare_output_only_single_nft";
+    setup(storage_path)?;
+
+    let wallet = make_wallet(storage_path, None, None).await?;
+    let account_0 = &create_accounts_with_funds(&wallet, 1).await?[0];
+    // Create second account without funds, so it only gets the NFT
+    let account_1 = wallet.create_account().finish().await?;
+    let addresses = &account_0.addresses().await?;
+    let account_0_address = addresses[0].address();
+    let addresses = &account_1.addresses().await?;
+    let account_1_address = addresses[0].address();
+
+    // Send NFT to second account
+    let tx = account_0
+        .mint_nfts([MintNftParams::new().try_with_address(account_1_address)?], None)
+        .await?;
+    account_0
+        .retry_transaction_until_included(&tx.transaction_id, None, None)
+        .await?;
+
+    let balance = account_1.sync(None).await?;
+    assert_eq!(balance.nfts().len(), 1);
+
+    println!("{:?}", balance);
+    let nft_data = &account_1.unspent_outputs(None).await?[0];
+    // Send NFT back to first account
+    let output = account_1
+        .prepare_output(
+            OutputParams {
+                recipient_address: *account_0_address,
+                amount: nft_data.output.amount(),
+                assets: Some(Assets {
+                    native_tokens: None,
+                    nft_id: Some(*balance.nfts().first().unwrap()),
+                }),
+                features: None,
+                unlocks: None,
+                storage_deposit: None,
+            },
+            None,
+        )
+        .await?;
+    let tx = account_1.send_outputs([output], None).await?;
+    account_1
+        .retry_transaction_until_included(&tx.transaction_id, None, None)
+        .await?;
+
```

Comment:
Is there anything we could assert here ?

Reply:
yea maybe something like total balance (not available) on account 1 to ensure no coins were given

Reply:
Thanks, already wondered why it passes, but then forgot to update :carlos:

Reply:
Added asserts